### PR TITLE
Util/get appointment by

### PR DESCRIPTION
**Pull request by evbambly**
getAppointmentById added, utilizing the original getAppointmentsByIds.

getAppointmentByIdOrThrow is the safe version, 
throwing instead of returning undefined, to standardize the resulting error.

### CODE REVIEW

File: packages/functions/src/utils/DbAppointmentUtils.ts [hunk 1]
```diff
@@ -120,6 +121,25 @@ export function toBookedAppointmentSync(
   }
 }
 
+export async function getAppointmentById(
```
Comment (yaron1m) [inline packages/functions/src/utils/DbAppointmentUtils.ts:124]: These 2 functions should be in the Appointment DAL, not here

File: packages/functions/src/utils/DbAppointmentUtils.ts [hunk 2]
```diff
@@ -120,6 +121,25 @@ export function toBookedAppointmentSync(
...
+  appointmentId: string
+): Promise<DbAppointment> {
```
Comment (yaron1m) [inline packages/functions/src/utils/DbAppointmentUtils.ts:126]: If this doesn't throw, it should return `DbAppointment | undefined`

File: packages/functions/src/utils/DbAppointmentUtils.ts (hunk 1 above)
Comment (evbambly) [reply packages/functions/src/utils/DbAppointmentUtils.ts:124]: Done

File: packages/functions/src/utils/DbAppointmentUtils.ts (hunk 2 above)
Comment (evbambly) [reply packages/functions/src/utils/DbAppointmentUtils.ts:126]: Done

Comment (evbambly) [reply packages/functions/src/utils/DbAppointmentUtils.ts:126]: None of the functions are built to handle undefined,
I removed the unsafe method

Comment (yaron1m) [reply packages/functions/src/utils/DbAppointmentUtils.ts:126]: Great

File: packages/functions/src/dal/AppointmentDataAccessLayer.ts [hunk 3]
```diff
@@ -26,6 +26,19 @@ export async function getAppointmentsByIds(
...
 }
 
+export async function getAppointmentByIdOrThrow(
+  appointmentId: string
+): Promise<DbAppointment> {
+  const foundAppointments = await getAppointmentsByIds([appointmentId]);
+  const appointment = foundAppointments[0];
+
+  if (!appointment) {
+    throw new Error("Appointment not found");
```
Comment (yaron1m) [inline packages/functions/src/dal/AppointmentDataAccessLayer.ts:36]: Consider adding the appointment id to the error message

**Review by yaron1m — APPROVED** [review]

File: packages/functions/src/dal/AppointmentDataAccessLayer.ts (hunk 3 above)
Comment (evbambly) [reply packages/functions/src/dal/AppointmentDataAccessLayer.ts]: Done

**Pull request merged**